DANS-KNAW/dans-dataverse-client-lib
# Support JSON output from Dataverse V6.2 plus retention period feature

Fixes DD-1565

And to be more specific; Publish of a Dataset will fail on 6.2 without this fix. 

# Description of changes
Support JSON output from Dataverse V6.2 plus retention period feature
Part of upgrading our Data Stations to version 6.2 with added patches (epic DD-1547).

# How to test

Build this and the build a local version of `dd-vault-metadata` that uses this newer lib. 
This will result in `0.27.1-SNAPSHOT`. 

Use that SNAPSHOT of the locally built client lib in `dd-vault-metadata`, `dd-validate-dans-bag` and `dd-ingest-flow`.
```
        <dependency>
            <groupId>nl.knaw.dans</groupId>
            <artifactId>dans-dataverse-client-lib</artifactId>
            <version>0.27.1-SNAPSHOT</version>
        </dependency>
```
Note: when building the client-lib now it produces version `0.27.1-SNAPSHOT`, but after releasing it will be `0.28.0`.

To test the backwards compatibility use a vagrant box that has our 6.0 version of datavesre; `start-preprovisioned-box.py -s` will give a working dev_archaeology basebox. 

Build the `dd-vault-metadata`, `dd-validate-dans-bag` and `dd-ingest-flow` services with that new client lib. 
Then deploy them on the vagrant box with `deploy.py -m <service name here> dev_archaeology`. 

The client lib (and the `dd-vault-metadata` service that uses it should be backwards compatible, we should test it by verifying that Publish also works on the Dataverse 6.0 that we have running now (also the DANS fork on our vagrant boxes). 

Testing the updated services can be done by running a SWORD2 example:
```
cd ~/git/dans-core-systems/modules/dd-dans-sword2-examples
mvnci
./run-simple-deposit.sh https://dev.sword2.archaeology.datastations.nl/collection/1 user001 user001  src/main/resources/example-bags/valid/default-open

./run-validation.sh https://dev.sword2.archaeology.datastations.nl/validate-dans-bag src/main/resources/example-bags/valid/default-open
```

The PR on `dans-core-systems` below explains how you can have Dataverse 6.2 on the vagrant box. Initially it is fine to test with the IQSS version of 6.2, without the RetentionPeriod feature. 

After you make sure the Publish works, you can go on and install the DANS patched version that has support for the RetentionPeriod feature. In this case add a retention period to a file and then Publish the Datasset. 

# Related PRs

(Add links)

*https://github.com/DANS-KNAW/dans-core-systems/pull/30
*https://github.com/IQSS/dataverse/pull/10336

# Notify

@DANS-KNAW/core-systems
